modules/300-prometheus/hooks/generate_istio_mtls_cert.go

```diff
@@ -0,0 +1,214 @@
+/*
+Copyright 2022 Flant JSC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package hooks
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"time"
+
+	"github.com/flant/addon-operator/pkg/module_manager/go_hook"
+	"github.com/flant/addon-operator/sdk"
+	"github.com/flant/shell-operator/pkg/kube_events_manager/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+
+	"github.com/deckhouse/deckhouse/go_lib/certificate"
+)
+
+const (
+	mTLSCrtPath = "prometheus.internal.prometheusScraperIstioMTLS.certificate"
+	mTLSKeyPath = "prometheus.internal.prometheusScraperIstioMTLS.key"
+)
+
+var _ = sdk.RegisterFunc(&go_hook.HookConfig{
+	Queue: "/modules/prometheus/generate_istio_mtls_cert",
+	Kubernetes: []go_hook.KubernetesConfig{
+		{
+			Name:       "istio_secret_ca",
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			FilterFunc: applySecertIstioCAFilter,
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-istio"},
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"cacerts"},
+			},
+		},
+		{
+			Name:       "secret_istio_mtls",
+			ApiVersion: "v1",
+			Kind:       "Secret",
+			FilterFunc: applySecertIstioMTLSFilter,
+			NamespaceSelector: &types.NamespaceSelector{
+				NameSelector: &types.NameSelector{
+					MatchNames: []string{"d8-monitoring"},
+				},
+			},
+			NameSelector: &types.NameSelector{
+				MatchNames: []string{"prometheus-scraper-istio-mtls"},
+			},
+		},
+	},
+	Schedule: []go_hook.ScheduleConfig{
+		{Name: "cron", Crontab: "0 0 * * *"}, // every day at 00:00 am
```

Comment:
We usually use magical 42 4 for it.
https://github.com/deckhouse/deckhouse/blob/7e8fe14a81468cce4ddb43ad22b4ca1aa0efa7ae/go_lib/hooks/tls_certificate/order_certificate.go#L143

Reply:
fixed